src/scripts/TranscriptsTrack.js

```diff
@@ -726,45 +1199,757 @@ const TranscriptsTrack = (HGC, ...args) => {
     }
 
     adjustTrackHeight() {
+
       this.computeTrackHeight();
 
+      if (!this.options.isVisible) this.trackHeight = 0;
+
+      this.allowResizeParentDiv = (this.options.allowResizeParentDiv) ? this.options.allowResizeParentDiv : true;
+
       if (this.trackHeightOld === this.trackHeight) {
         return false;
       }
 
       this.pubSub.publish("trackDimensionsModified", {
         height: this.trackHeight,
-        resizeParentDiv: true,
+        resizeParentDiv: this.allowResizeParentDiv,
         trackId: this.trackId,
         viewId: this.viewId
       });
 
+      if (this.trackHeightOld === 0) this.rerender(this.options, true);
+
       return true;
     }
 
     formatTranscriptData(ts) {
       const strand = ts[5];
-      const stopCodonPos = ts[12] === "." ? "." : (strand === "+" ? +ts[12] + 2 : +ts[12] - 1);
-      const startCodonPos = ts[11] === "." ? "." : (strand === "+" ? +ts[11] - 1 : +ts[11] + 2);
+      let startCodonPos = ts[11] === "." ? "." : (strand === "+" ? +ts[11] - 1 : +ts[11] + 2);
+      let stopCodonPos = ts[12] === "." ? "." : (strand === "+" ? +ts[12] + 2 : +ts[12] - 1);
       const exonStarts = ts[9].split(",").map((x) => +x - 1);
       const exonEnds = ts[10].split(",").map((x) => +x);
       const txStart = +ts[1] - 1;
       const txEnd = +ts[2] - 1;
+      const strandedStartCodonPos = ((startCodonPos !== ".") && (stopCodonPos !== ".")) ? ((strand === "+") ? ((startCodonPos < stopCodonPos) ? startCodonPos : stopCodonPos) : ((startCodonPos > stopCodonPos) ? startCodonPos : stopCodonPos)) : ".";
+      const strandedStopCodonPos = ((startCodonPos !== ".") && (stopCodonPos !== ".")) ? ((strand === "+") ? ((startCodonPos < stopCodonPos) ? stopCodonPos : startCodonPos) : ((startCodonPos > stopCodonPos) ? stopCodonPos : startCodonPos)) : ".";
+      const strandedTxStart = (strand === "+") ? ((txStart < txEnd) ? txStart : txEnd) : ((txStart > txEnd) ? txStart : txEnd);
+      const strandedTxEnd = (strand === "+") ? ((txStart < txEnd) ? txEnd : txStart) : ((txStart > txEnd) ? txEnd : txStart);
+      
+      let boxBlockCount = -1;
+      let boxBlockLengths = [];
+      let boxBlockStarts = [];
+      if (this.options.blockStyle === "boxplot") {
+        if (!startCodonPos) { startCodonPos = +ts[6] }
+        if (!stopCodonPos) { stopCodonPos = +ts[7] }
+        boxBlockCount = +ts[9];
+        boxBlockLengths = ts[10].split(",").map((x) => +x);
+        boxBlockStarts = ts[11].split(",").map((x) => +x);
+      }
+
+      function calculateBlocks() {
```

Comment:
Oh boy... I didn't check this function line by line and assume you got it right :-)

Reply:
I used Gencode hg19, hg38, and mm10 annotations to examine all the different scenarios I could picture would need rendering (transcripts with and without UTR, different strands, etc.).